mayastor/tests/error_fault_child_total.rs

```diff
@@ -0,0 +1,192 @@
+        MayastorEnvironment,
+        Reactor,
+    },
+    subsys::Config,
+};
+
+use spdk_sys::{
+    create_aio_bdev,
+    spdk_vbdev_error_create,
+    spdk_vbdev_error_inject_error,
+    SPDK_BDEV_IO_TYPE_READ,
+    SPDK_BDEV_IO_TYPE_WRITE,
+};
+
+static ERROR_COUNT_TEST_NEXUS: &str = "error_fault_child_test_nexus";
+
+static DISKNAME1: &str = "/tmp/disk1.img";
+static BDEVNAME1: &str = "aio:///tmp/disk1.img?blk_size=512";
+
+static DISKNAME2: &str = "/tmp/disk2.img";
+
+static ERROR_DEVICE: &str = "error_device";
+static EE_ERROR_DEVICE: &str = "EE_error_device"; // The prefix is added by the vbdev_error module
+static BDEV_EE_ERROR_DEVICE: &str = "bdev:///EE_error_device";
+
+static YAML_CONFIG_FILE: &str = "/tmp/error_fault_child_test_nexus.yaml";
+
+// constant used by the vbdev_error module but not exported
+const VBDEV_IO_FAILURE: u32 = 1;
+
+#[test]
+fn nexus_fault_child_read_test() {
+    common::truncate_file(DISKNAME1, 64 * 1024);
+    common::truncate_file(DISKNAME2, 64 * 1024);
+
+    let mut config = Config::default();
+    config.err_store_opts.enable_err_store = true;
+    config.err_store_opts.err_store_size = 256;
+    config.err_store_opts.fault_child_on_error_count = true;
+    config.err_store_opts.max_error_age_ns = 1_000_000_000;
+    config.err_store_opts.max_total_errors = 4;
+
+    config.write(YAML_CONFIG_FILE).unwrap();
+
+    test_init!(Some(YAML_CONFIG_FILE.to_string()));
+
+    Reactor::block_on(async {
+        create_error_bdev().await;
+        create_nexus().await;
+
+        check_nexus_state_is(NexusStatus::Online);
+
+        inject_error(SPDK_BDEV_IO_TYPE_READ, VBDEV_IO_FAILURE, 10).await;
+        inject_error(SPDK_BDEV_IO_TYPE_WRITE, VBDEV_IO_FAILURE, 10).await;
+
+        for _ in 0 .. 3 {
+            err_read_nexus_both(false).await;
+            reactor_run_millis(1);
+        }
+        for _ in 0 .. 2 {
+            // the second iteration causes the error count to exceed the max no
+            // of retry errors (4) for the read and causes the child to be
+            // removed
+            err_read_nexus_both(false).await;
+            reactor_run_millis(1);
+        }
+    });
+
+    // error child should be removed from the IO path here
+
+    check_nexus_state_is(NexusStatus::Degraded);
+
+    Reactor::block_on(async {
+        err_read_nexus_both(true).await; // should succeed because both IOs go to the remaining child
+        err_write_nexus(true).await; // should succeed because the IO goes to
+                                     // the remaining child
+    });
+
+    mayastor_env_stop(0);
+
+    common::delete_file(&[DISKNAME1.to_string()]);
+    common::delete_file(&[DISKNAME2.to_string()]);
+    common::delete_file(&[YAML_CONFIG_FILE.to_string()]);
+}
+
+fn check_nexus_state_is(expected_status: NexusStatus) {
+    let nexus = nexus_lookup(ERROR_COUNT_TEST_NEXUS).unwrap();
+    assert_eq!(nexus.status(), expected_status);
+}
+
+async fn create_error_bdev() {
```

Comment:
There is a common version of this in common/error_bdev.rs.

Reply:
Now using the common version of this function (and inject_error()) in this and error_fault_child_retries.rs